Automattic/wp-calypso
# Agency Dashboard: Add Request to Check Sites Connection Health

Currently, the connection health status is populated in the site profiles index during an audit, which means that there might be a significant delay between the time site went down and the index update. More context can be found pedEI5-2s-p2 here.

In WordPress.com, we use the `/test-connection` endpoint that makes an API request to the remote site to verify the connection's health. 

The API was changed so we can use this endpoint also on Calypso green.

**Backend task**: 1202619025189113-as-1203022769710885

#### Proposed Changes

- In the Calypso dashboard, the connection status of each site will be fetched.
- If the connection status of the site is not healthy, an error message will appear to the user.

#### Testing Instructions

- Make sure to have a partner account.
- Create an ephemeral site
- Connect Jetpack
- Ensure the site is now visible in the [Agency Dashboard](https://calypso.live/?image=registry.a8c.com/calypso/app:build-45797&env=jetpack)
- Delete the ephemeral site
- Visit the site to ensure it is gone
- Refresh the [Agency Dashboard](https://calypso.live/?image=registry.a8c.com/calypso/app:build-45797&env=jetpack) page
- Verify that there is now an indication about the unhealthy site connection

#### Screenshots

![image](https://user-images.githubusercontent.com/6406071/194975504-f0915ca4-eac8-4ab5-8b1d-c89ce77e88be.png)


#### Pre-merge Checklist






Related to 1202619025189113-as-1203022769710885
